nixos/modules/system/boot/stage-1-init.sh

```diff
@@ -116,19 +111,40 @@ source @earlyMountScript@
 mkdir -p /tmp
 mkfifo /tmp/stage-1-init.log.fifo
 logOutFd=8 && logErrFd=9
+
+# askPassword <prompt>
+askPassword() {
+    # Ensure that we don't reset the echo flag.
+    curStatus=`stty -g`
+    stty -echo
+
+    printf "%s" "$1" >&2
+    IFS= read password
```

Comment:
Also, ShellCheck complains that "read without -r will mangle backslashes" [1] -- maybe this should be changed so users with backslashes in their passwords don't get bitten?

[1] https://www.shellcheck.net/wiki/SC2162